src/main/java/de/tum/in/www1/artemis/service/connectors/AtheneService.java

```diff
@@ -0,0 +1,288 @@
+    }
+
+    /**
+     * Calls the remote Athene service to submit a Job for calculating automatic feedback
+     * @param exercise the exercise the automatic assessments should be calculated for
+     */
+    public void submitJob(TextExercise exercise) {
+        submitJob(exercise, 1);
+    }
+
+    /**
+     * Calls the remote Athene service to submit a Job for calculating automatic feedback
+     * Falls back to naive splitting for less than 10 submissions
+     * Note: See `TextSubmissionService:getTextSubmissionsByExerciseId` for selection of Submissions.
+     * @param exercise the exercise the automatic assessments should be calculated for
+     * @param maxRetries number of retries before the request will be canceled
+     */
+    public void submitJob(TextExercise exercise, int maxRetries) {
+        log.debug("Start Athene Service for Text Exercise \"" + exercise.getTitle() + "\" (#" + exercise.getId() + ").");
+
+        // Find all submissions for Exercise
+        List<TextSubmission> textSubmissions = textSubmissionService.getTextSubmissionsByExerciseId(exercise.getId(), true, false);
+
+        // We only support english languages so far, to prevent corruption of the clustering
+        textSubmissions.removeIf(textSubmission -> textSubmission.getLanguage() != Language.ENGLISH);
+
+        // Athene only works if more than 10 submissions are available
+        // else textBlockService is used
+        if (textSubmissions.size() >= 10) {
+
+            log.info("Calling Remote Service to calculate automatic feedback for " + textSubmissions.size() + " submissions.");
+
+            try {
+                final Request request = new Request(exercise.getId(), textSubmissions, ARTEMIS_SERVER_URL + ATHENE_RESULT_API_PATH + exercise.getId());
+                Response response = connector.invokeWithRetry(API_ENDPOINT, request, authorizationHeaderForSymmetricSecret(API_SECRET), maxRetries);
+                log.info("Remote Service to calculate automatic feedback responded: " + response.detail);
+
+                // Register task for exercise as running, AtheneResource calls finishTask on result receive
+                startTask(exercise.getId());
+            }
+            catch (NetworkingError networkingError) {
+                log.error("Error while calling Remote Service", networkingError);
+            }
+
+        }
+        else {
+
+            log.info("More than 10 submissions needed to calculate automatic feedback. Falling back to naive splitting");
+
+            List<TextBlock> set = new ArrayList<>();
+
+            // Split Submissions into Blocks
+            for (TextSubmission textSubmission : textSubmissions) {
+
+                final List<TextBlock> blocks = textBlockService.splitSubmissionIntoBlocks(textSubmission);
+                textSubmission.setBlocks(blocks);
+                set.addAll(blocks);
+
+            }
+
+            textBlockRepository.saveAll(set);
+
+        }
```

Comment:
I think we can even skip this part. This should be done before assessing each submission. so only exit if there are < 10